barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/ultra_circuit_builder.hpp

```diff
@@ -211,12 +211,31 @@ class UltraCircuitBuilder_ : public CircuitBuilderBase<typename Arithmetization:
             if (a < other.a) {
                 return true;
             }
-            if (a == other.a) {
-                if (b < other.b) {
-                    return true;
-                }
+            if (other.a < a) {
+                return false;
+            }
+            if (b < other.b) {
+                return true;
+            }
+            if (other.b < b) {
+                return false;
+            }
+
+            if (uint256_t(lo_0) < uint256_t(other.lo_0)) {
+                return true;
             }
-            return false;
```

Comment:
Do we need the checks on `lo_0, hi_0, hi_1`? These quantities are derived from `a` and `b`. So if `a`, `b` match, `lo/hi` will match.

<img width="570" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/1476668/f28d1b25-110d-4867-b9bb-51055610ac16">